test/integration/daemonset/daemonset_test.go

```diff
@@ -52,14 +54,21 @@ import (
 var zero = int64(0)
 
 func setup(t *testing.T) (context.Context, kubeapiservertesting.TearDownFunc, *daemon.DaemonSetsController, informers.SharedInformerFactory, clientset.Interface) {
-	// Disable ServiceAccount admission plugin as we don't have serviceaccount controller running.
-	server := kubeapiservertesting.StartTestServerOrDie(t, nil, []string{"--disable-admission-plugins=ServiceAccount,TaintNodesByCondition"}, framework.SharedEtcd())
+	return setupWithServerSetup(t, framework.TestServerSetup{
+		ModifyServerRunOptions: func(opts *options.ServerRunOptions) {
+			opts.Admission.GenericAdmission.DisablePlugins = []string{
```

Comment:
I expect that we want to ensure these plugins are disabled in all the tests  (see https://github.com/kubernetes/kubernetes/pull/110229). Can move this initialization into the `setupWithServerSetup` and merge it with the `serverSetup` param

Reply:
I think we can't merge just the `DisabledPlugins` option but end up merging `ModifyServerRunOptions`, one of the two options in `TestServerSetup`. How about adding newDefaultTestServerSetup so we can use it if we want to?

Reply:
Or maybe we can call the existing `ModifyServerRunOptions` function after our default `ModifyServerRunOptions`.

```go
if serverSetup.ModifyServerRunOptions == nil {                               
  serverSetup.ModifyServerRunOptions = func(opts *options.ServerRunOptions) {
    opts.Admission.GenericAdmission.DisablePlugins = []string{               
      // Disable ServiceAccount admission plugin as we don't have            
      // serviceaccount controller running.                                  
      "ServiceAccount",                                                      
      "TaintNodesByCondition",                                               
    }                                                                        
  }                                                                          
} else {                                                                     
  serverSetup.ModifyServerRunOptions = func(opts *options.ServerRunOptions) {
    opts.Admission.GenericAdmission.DisablePlugins = []string{               
      // Disable ServiceAccount admission plugin as we don't have            
      // serviceaccount controller running.                                  
      "ServiceAccount",                                                      
      "TaintNodesByCondition",                                               
    }                                                                        
    serverSetup.ModifyServerRunOptions(opts)                                 
  }                                                                          
}                                                                            

clientSet, config, closeFn := framework.StartTestServer(t, serverSetup)
...
```

Do you think this is the better way?

Reply:
makes sense. done.